pkg/maven/maven.go

```diff
@@ -107,6 +134,146 @@ func Evaluate(options *EvaluateOptions, expression string, command mavenExecRunn
 	return value, nil
 }
 
+// InstallFile installs a maven artifact and its pom into the local maven repository.
+// If "file" is empty, only the pom is installed. "pomFile" must not be empty.
+func InstallFile(file, pomFile string, command mavenExecRunner) error {
+	if len(pomFile) == 0 {
+		return fmt.Errorf("pomFile can't be empty")
+	}
+
+	var defines []string
+	if len(file) > 0 {
+		defines = append(defines, "-Dfile="+file)
+	} else {
+		defines = append(defines, "-Dfile="+pomFile)
+	}
+	defines = append(defines, "-DpomFile="+pomFile)
+	mavenOptionsInstall := ExecuteOptions{
+		Goals:   []string{"install:install-file"},
+		Defines: defines,
+		PomPath: pomFile,
+	}
+	_, err := Execute(&mavenOptionsInstall, command)
+	if err != nil {
+		return fmt.Errorf("failed to install maven artifacts: %w", err)
+	}
+	return nil
+}
+
+// InstallMavenArtifacts finds maven modules (identified by pom.xml files) and installs the artifacts into the local maven repository.
+func InstallMavenArtifacts(command mavenExecRunner, options EvaluateOptions) error {
+	return doInstallMavenArtifacts(command, options, newUtils())
+}
+
+func doInstallMavenArtifacts(command mavenExecRunner, options EvaluateOptions, utils mavenUtils) error {
+	err := flattenPom(command)
+	if err != nil {
+		return err
+	}
+
+	pomFiles, err := utils.glob(filepath.Join("**", "pom.xml"))
+	if err != nil {
+		return err
+	}
+
+	oldWorkingDirectory, err := utils.getwd()
+	if err != nil {
+		return err
+	}
+
+	for _, pomFile := range pomFiles {
+		log.Entry().Info("Installing maven artifacts from module: " + pomFile)
+		dir := path.Dir(pomFile)
+		err = utils.chdir(dir)
+		if err != nil {
+			return err
+		}
+
+		// Set pom path fix here because we cd'ed into the module's directory, so the pom is not in a subdirectory
+		options.PomPath = "pom.xml"
+		packaging, err := Evaluate(&options, "project.packaging", command)
+		if err != nil {
+			return err
+		}
+
+		if packaging == "pom" {
+			err = InstallFile("", "pom.xml", command)
+			if err != nil {
+				return err
+			}
+		} else {
+			err = installJarWarArtifacts(command, utils, options)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Why don't you special case pom-packaging, and otherwise use the packaging to know the target artifacts? I mean, why do you ignore packaging if it is not "pom", and test for a "jar" or "war" only?